$NetBSD: patch-config.lib,v 1.3 2020/08/18 03:27:49 tnn Exp $

Compile 'strip' test with CFLAGS, so -isysroot can be passed on OSX.
Correct library detection on NetBSD.
On Darwin, be able to locate iconv.h inside the SDK.
Use pkg-config to detect liblzo2 as hand-crafted method seems unreliable.

--- config.lib.orig	2020-06-01 21:22:01.000000000 +0000
+++ config.lib
@@ -2088,7 +2088,7 @@ check_strip() {
 	elif [ "$os" = "OSX" ]; then
 		# Most targets have -V in strip, to see if they exists... OSX doesn't.. so execute something
 		echo "int main(int argc, char *argv[]) { }" > strip.test.c
-		$cxx_host strip.test.c -o strip.test
+		$cc_host $CFLAGS strip.test.c -o strip.test
 		check_compiler "host strip" "strip" "$host" "$strip" "$STRIP" "strip" "strip" "3" "strip.test"
 		rm -f strip.test.c strip.test
 	else
@@ -2633,7 +2633,7 @@ detect_library() {
 		if [ -z "$res" ]; then
 			log 2 "  trying /opt/local/include/$4$5... no"
 		fi
-		if [ -z "$res" ] && [ "$os" = "NETBSD" ]; then
+		if [ -z "$res" ]; then
 			eval "$2=`ls -1 /usr/pkg/include/$4*.h 2>/dev/null | egrep \"\/$5\$\"`"
 			eval "res=\$$2"
 			if [ -z "$res" ]; then
@@ -2725,7 +2725,7 @@ detect_zlib() {
 }
 
 detect_lzo2() {
-	detect_library "$with_lzo2" "lzo2" "liblzo2.a" "lzo/" "lzo1x.h"
+	detect_pkg_config "$with_lzo2" "lzo2" "lzo2" "2.10"
 }
 
 detect_fluidsynth() {
@@ -2885,6 +2885,9 @@ detect_iconv() {
 				break
 			fi
 		done
+		if [ -z "$iconv" ] && [ "$os" = "OSX" ]; then
+			iconv=`xcrun --show-sdk-path`/usr/include/iconv.h
+		fi
 	else
 		# Make sure it exists
 		iconv=`ls $with_iconv/include/iconv.h 2>/dev/null`
